File: backend/appointments/src/main/java/com/medical/homevisits/appointments/prescription/entity/Prescription.java
package com.medical.homevisits.appointments.prescription.entity;

import com.medical.homevisits.appointments.appointment.entity.Appointment;
import com.medical.homevisits.appointments.appointment.entity.AppointmentStatus;
import com.medical.homevisits.appointments.doctor.entity.Doctor;
import com.medical.homevisits.appointments.patient.entity.Patient;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.UUID;
@Entity
@Getter
@Setter
public class Prescription {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne
    @JoinColumn(name = "doctor_id") 
    private Doctor doctor;

    @ManyToOne
    @JoinColumn(name = "patient_id")
    private Patient patient;

    private PrescriptionStatus status; //status of the appointment (Available, In_progress, Completed)
    private LocalDateTime PrescriptionTime;
    
    private String medication;
    private String dosage;
    private String notes;
}

File: frontend/app/src/main/java/com/medical/wizytydomowe/api/prescriptions/Prescription.kt
package com.medical.wizytydomowe.api.prescriptions

import com.medical.wizytydomowe.api.users.Doctor
import com.medical.wizytydomowe.api.users.Patient
import java.io.Serializable

data class Prescription(
    val id: String?,
    val doctor: Doctor?,
    val patient: Patient?,
    val date: String?,
    val medication: String?,
    val dosage: String?,
    val notes: String?
) : Serializable
